vitrina/datasets/managers.py

```diff
@@ -74,6 +77,43 @@ def _filter_datasets_for_user(self, user: "User", datasets: QuerySet["Dataset"])
         for ds_path in represented_dataset_paths:
             accessible_filter |= Q(path__startswith=ds_path)
 
+        info_system_orgs = Representative.objects.filter(
+            content_type=org_ct, user_id=user.id, information_system_representative=True
+        ).values_list("object_id", flat=True)
+
+        if info_system_orgs.exists():
```

Comment:
```suggestion
        if info_system_orgs.all():
```
should save a query here, as you are selecting anyways if true.